### PR TITLE
Opp design practice

### PR DESCRIPTION
**Pull request by maiklgravi**

### CODE REVIEW

File: app/Providers/AppServiceProvider.php [hunk 1]
```diff
@@ -14,7 +17,13 @@ class AppServiceProvider extends ServiceProvider
      */
     public function register()
     {
-        //
+        $this->app->bind(RequestActivityLoggerInterface::class, function () {
+            if (App::environment('production')) {
+                return $this->app->make(RequestActivityLoggerInterface::class);
+            } else {
+                return $this->app->make(DummyRequestActivityLogger::class);
```
Comment (egorbwork) [inline app/Providers/AppServiceProvider.php:24]: DebugRequestActivityLogger was requested to be injected.

File: app/Console/Commands/RussionRouletteCommand.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+class RussionRouletteCommand extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'game:name';
+
```
Comment (egorbwork) [inline app/Console/Commands/RussionRouletteCommand.php:17]: please use more adequate name for console command

File: app/Console/Commands/RussionRouletteCommand.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'RussionRoulette game were youser win or lose.';
```
Comment (egorbwork) [inline app/Console/Commands/RussionRouletteCommand.php:23]: please provide definition for `youser` word.

File: app/Console/Commands/RussionRouletteCommand.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ 
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        
+        $user = $this->ask("What you user id?");
```
Comment (egorbwork) [inline app/Console/Commands/RussionRouletteCommand.php:45]: что ты пользователь айди?

File: app/Console/Commands/RussionRouletteCommand.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        $win = $this->cacheRepository->get('WinStatistic', []);
```
Comment (egorbwork) [inline app/Console/Commands/RussionRouletteCommand.php:46]: I recommend to use `statistics` word everywhere

File: app/Console/Commands/RussionRouletteCommand.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        $lose = $this->cacheRepository->get('loseStatistic', []);
+        $win[$user] = $win[$user] ?? 0;
+        $lose[$user] = $lose[$user] ?? 0;
+        $randombullet = random_int(1,6);
```
Comment (egorbwork) [inline app/Console/Commands/RussionRouletteCommand.php:50]: Please keep for statistics purpose the number of how many times which place for bullet were randomized.

File: app/Console/Commands/ShowStatusCommand.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+use Illuminate\Contracts\Config\Repository as ConfigRepository;
+
+class ShowStatusCommand extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'status:name';
```
Comment (egorbwork) [inline app/Console/Commands/ShowStatusCommand.php:16]: Please use more adequate name.

**Review by egorbwork — CHANGES_REQUESTED** [review]
It is not ready.

File: app/Console/Commands/ShowStatusCommand.php (hunk 7 above)
Comment (maiklgravi) [reply app/Console/Commands/ShowStatusCommand.php]: I rename this

File: app/Console/Commands/RussionRouletteCommand.php (hunk 4 above)
Comment (maiklgravi) [reply app/Console/Commands/RussionRouletteCommand.php]: i rewrite this

File: app/Console/Commands/RussionRouletteCommand.php (hunk 3 above)
Comment (maiklgravi) [reply app/Console/Commands/RussionRouletteCommand.php]: i change this

File: app/Console/Commands/RussionRouletteCommand.php (hunk 2 above)
Comment (maiklgravi) [reply app/Console/Commands/RussionRouletteCommand.php]: i rename this

File: app/Providers/AppServiceProvider.php (hunk 1 above)
Comment (maiklgravi) [reply app/Providers/AppServiceProvider.php]: my mistake i chanege this

File: app/Console/Commands/RussionRouletteCommand.php (hunk 6 above)
Comment (maiklgravi) [reply app/Console/Commands/RussionRouletteCommand.php:44]: i add this

File: app/Providers/AppServiceProvider.php [hunk 8]
```diff
@@ -14,7 +17,13 @@ class AppServiceProvider extends ServiceProvider
      */
     public function register()
     {
-        //
+        $this->app->bind(RequestActivityLoggerInterface::class, function () {
+            if (App::environment('production')) {
+                return $this->app->make(RequestActivityLoggerInterface::class);
```
Comment (egorbwork) [inline app/Providers/AppServiceProvider.php:22]: ProductionRequestActivityLogger also still not used.

File: app/Console/Commands/RussionRouletteCommand.php [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+     * @var string
+     */
+    protected $signature = 'roullette:game';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'RussionRoulette game were user win or lose.';
```
Comment (egorbwork) [inline app/Console/Commands/RussionRouletteCommand.php:23]: maybe `where`?

File: app/Console/Commands/ShowStatusCommand.php [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+     * @var string
+     */
+    protected $description = 'Show status for win,lose and all Bullet position';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct(CacheRepository $cacheRepository, ConfigRepository $configRepository)
```
Comment (egorbwork) [inline app/Console/Commands/ShowStatusCommand.php:30]: You don't need config repository in this command.

File: app/Console/Commands/ShowStatusCommand.php [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+        foreach ($statisticLose as $key => $count ) {
+            $table[] = [$key, $count ];
+        }
+        $this->table(['user', 'statisticLose' ], $table);
+        $table = [];
+        foreach ($statisticBulletPosition as $key => $count ) {
+            $table[] = [$key, $count ];
+        }
+        $this->table(['randombullet', 'statisticBulletPosition'],$table);
+        $this->cacheRepository->set('WinStatistic', $statisticWin);
```
Comment (egorbwork) [inline app/Console/Commands/ShowStatusCommand.php:62]: Why you are setting anything in cache here?

File: app/Console/Commands/ShowStatusCommand.php [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+    public function handle()
+    {
+        $statisticWin = $this->cacheRepository->get('WinStatistic', []);
+        $statisticLose = $this->cacheRepository->get('LoseStatistic', []);
+        $statisticBulletPosition = $this->cacheRepository->get('BulletStatistic', []);
+        $table = [];
+        foreach ($statisticWin as $key => $count ) {
+            $table[] = [$key, $count ];
+        }
+        $this->table(['user', 'statisticWin' ], $table);
```
Comment (egorbwork) [inline app/Console/Commands/ShowStatusCommand.php:51]: an empty line before each table would be nice.

**Review by egorbwork — CHANGES_REQUESTED** [review]
Still exists mistakes.

File: app/Providers/AppServiceProvider.php (hunk 8 above)
Comment (maiklgravi) [reply app/Providers/AppServiceProvider.php:22]: yes my mistake

File: app/Console/Commands/RussionRouletteCommand.php (hunk 9 above)
Comment (maiklgravi) [reply app/Console/Commands/RussionRouletteCommand.php:23]: yes

File: app/Console/Commands/ShowStatusCommand.php (hunk 11 above)
Comment (maiklgravi) [reply app/Console/Commands/ShowStatusCommand.php:62]: my inattention

File: app/Console/Commands/ShowStatusCommand.php (hunk 12 above)
Comment (maiklgravi) [reply app/Console/Commands/ShowStatusCommand.php:52]: i aply this

**Review by egorbwork — APPROVED** [review]
Now is acceptable.

**Pull request merged**